server/src/com/mirth/connect/connectors/file/filesystems/SftpConnection.java

```diff
@@ -222,9 +222,18 @@ public List<String> listDirectories(String fromDir) throws Exception {
 
     @Override
     public boolean exists(String file, String path) {
+        if(null == file) {

```

Comment:
Should this also check null for path?

Reply:
Checking `path` for `null` wouldn't hurt, but I don't think it's strictly necessary in this case. Like you mentioned, checking `file` for `null` is mainly to avoid an NPE, which is not as issue with `path`.

There wouldn't  anything wrong with throwing an `IllegalArgumentException`, but since the existing behavior of the method is to catch all exceptions and return `false`, I lean toward just returning `false` to avoid any unforeseen bugs.

Summing up, I wouldn't make any changes here.

Reply:
@pladesma It's your product, so I'll do whatever you want, but I'd rather have an error thrown than for this check to return `false` and then have something very very weird happen later. I suppose "can't write file" is the likely next error you'll encounter if this method returns `false` because `path`/`file` are not sane.